integration_test/README.md

```diff
@@ -142,7 +142,39 @@ Then, inside `applications/<application>/`:
 1.  (if necessary) `exercise`. This is only needed
     sometimes, e.g. to get the application to log to a particular file.
 1.  (if you want to test logging) `expected_logs.yaml`
-1.  (if you want to test metrics) `metric_name.txt`
+1.  (if you want to test metrics) `expected_metrics.yaml`
+
+### expected_metrics.yaml
+
+We use `expected_metrics.yaml` both as a test artifact and as a source for documentation. All metrics ingested from the integration should be documented here.
+
+A sample `expected_metrics.yaml` snippet looks like:
+
+```yaml
+- type: workload.googleapis.com/apache.current_connections
+  value_type: INT64
+  kind: GAUGE
+  monitored_resource: gce_instance
+  labels:
+    server_name: .*
+  is_representative: true
+  optional: true
+```
+
+`type`, `value_type` and `kind` come directly from the metric descriptor for that metric. `monitored_resource` should always be `gce_instance`.
+
+`labels` is an exhaustive list of labels associated with the metric. Each key in `labels` is the label name, and its value is a regular expression. During the test, each label returned by the time series for that metric is checked against `labels`: every label in the time series must be present in `labels`, and its value must match the regular expression.
+
+For example, if a metric defines a label `operation` whose values can only be `read` or `write`, then an appropriate `labels` map in `expected_metrics.yaml` would be as follows:
+
+```yaml
+  labels:
+    state: (read|write)
```

Comment:
Are these parentheses required? If not, may be best to omit them, so we don't imply any capture groups.

Reply:
You're right, they don't seem to be needed per [RE2](https://github.com/google/re2/wiki/Syntax). Will fix.